spec.go

```diff
@@ -311,26 +311,47 @@ func createLibcontainerMount(cwd string, m specs.Mount) *configs.Mount {
 	}
 }
 
-func createCgroupConfig(name string, spec *specs.Spec) (*configs.Cgroup, error) {
+func createCgroupConfig(name string, useSystemdCgroup bool, spec *specs.Spec) (*configs.Cgroup, error) {
 	var (
 		err          error
 		myCgroupPath string
 	)
 
+	c := &configs.Cgroup{
+		Resources: &configs.Resources{},
+	}
+
 	if spec.Linux.CgroupsPath != nil {
 		myCgroupPath = libcontainerUtils.CleanPath(*spec.Linux.CgroupsPath)
+	}
+
+	if useSystemdCgroup {
+		if myCgroupPath == "" {
+			c.Parent = "system.slice"
+			c.ScopePrefix = "runc"
+			c.Name = name
+		} else {
+			// Parse the path from expected "slice:prefix:name"
+			// for e.g. "system.slice:docker:1234"
+			parts := strings.Split(myCgroupPath, ":")
```

Comment:
This is similar to how docker supported systemd cgroups earlier using a daemon option. When the native.cgroupdriver is set to systemd, then createSpec could special form cgroupsPath using separators.